package.json

```diff
@@ -6,20 +6,9 @@
   },
   "description": "JavaScript ROOT",
   "homepage": "https://root.cern/js/",
-  "type": "module",
-  "main": "./build/jsroot.js",
   "module": "./modules/main.mjs",
-  "exports": {
-    ".": "./modules/main.mjs",
-    "./core": "./modules/core.mjs",
-    "./draw": "./modules/draw.mjs",
-    "./io": "./modules/io.mjs",
-    "./tree": "./modules/tree.mjs",
-    "./colors": "./modules/base/colors.mjs",
-    "./hierarchy": "./modules/gui/HierarchyPainter.mjs",
-    "./latex": "./modules/base/latex.mjs",
-    "./geom": "./modules/geom/TGeoPainter.mjs"
-  },
+  "main": "./modules/main.mjs",
```

Comment:
Why you remove `"type": "module"` and all exports?
Last is required to be able import dedicated components like in https://github.com/root-project/jsroot/blob/master/demo/node/selector.js

Reply:
Because the `browser` field isn't respected if either `"type": "module"` or `"exports"` fields are set. This is because of precedence. At least in webpack.

That's why I am inclining towards the other solution which I mentioned in my last comment.

I didn't realize that the internal code also uses these exports.

Reply:
Will it work for you if one keep `"type": "module"` and all exports?
Rest looks fine for me

Reply:
That exactly make problem for you - `"type": "module"` or sub-exports?

In your code I see that you make direct use of import of node modules - which is not work for using of modules in  browser. Also you define `"browser"` parameter in package.json differently.

Reply:
> That exactly make problem for you - "type": "module" or sub-exports?

No, this one works fine with webpack without having to change other fields in package.json.

> In your code I see that you make direct use of import of node modules - which is not work for using of modules in browser. Also you define "browser" parameter in package.json differently.

It works with this approach. The `"browser"` field is used differently here in the sense that now it specifies exactly which imported modules only work with node and webpack automatically mocks these modules so they don't throw errors in the browser. This way, we don't have to use the rollup bundle/build and webpack automatically transpiles the ES modules.

I had to make the following changes to code so everything works:

1. Define in the `"browser"` field of package.json to specify which modules are node modules and are to be ignored by the bundler when bundling for web/browser.
2. For modules that use top-level await, directly import them through the ES import syntax.
    * This required making some changes in the code so the exclusion for qt5web works fine. (I hope it does)
3. Move dynamic import to a single place. (The `dynamicImport` function)
    * Dynamically importing node modules is fine, since webpack is going to ignore them anyway.
    * But dynamically importing arbitrary files/URLs throws warnings in webpack. So I had to make this change to narrow down the warning to a single place so it's easier to adapt/change later.
       ![image](https://user-images.githubusercontent.com/36920441/175915466-490e9608-e2ed-4bbb-9be4-f67496ccbd78.png)
       This is the only warning that remains and I am not able to fix it yet.